ruby_oop/hangman/lib/random_word.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class RandomWord
+  # this class should create a word object that is for a random
+  # word from the text file in assets/english_words.txt
+  attr_accessor :rand_word
+
+  def initialize
+    @rand_word = random_word
+  end
+
+  def random_word
+    loop do
+      @rand_word = File.readlines('../assets/english_words.txt').sample
+      break if @rand_word.length.between?(6, 12)
+    end
```

Comment:
`File.readlines` is a relatively expensive operation. And in this case we aren't expecting it to return different results each iteration, so we can pull it out of the `loop`.

```ruby
word_list = File.readlines('...')
loop do
  word = word_list.sample
  # ...
end
```